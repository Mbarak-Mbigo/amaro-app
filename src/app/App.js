import React from 'react';
import styled, { createGlobalStyle } from 'styled-components'

import { SILVER_GREY } from '../utils/colors';

const GlobalStyle = createGlobalStyle`
  body {
    background-color: ${SILVER_GREY};
    margin: 0;
  }
`
const Container = styled.div``;

const Header = styled.header`
  height: 12rem;
  display: flex;
  justify-content: center;
  align-items: center;
  border: 2px solid black;
`

const MainContent = styled.section`
  display: flex;
  justify-content: center;
  border: 2px solid red;
  height: calc(100vh - 288px)
`;

const Footer = styled.footer`
  border: 2px solid green;
  height: 98px;
`;

const App = () => (
      <Container>
      <GlobalStyle />
      <Header>Header Part</Header>
      <MainContent>
        App
      </MainContent>
      <Footer>
        Footer
      </Footer>
      </Container>
  );

export default App;
