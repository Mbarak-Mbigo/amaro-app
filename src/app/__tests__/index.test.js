import React from 'react';

import renderer from 'react-test-renderer';

import App from '..';

describe('App Component Tests', () => {
  const component = renderer.create(<App />);

  test('renders App component', () => {
    expect(component.toJSON()).toMatchSnapshot();
  })
})
