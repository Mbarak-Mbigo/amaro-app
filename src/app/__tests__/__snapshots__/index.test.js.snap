// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders App component 1`] = `
<div
  className="sc-bdVaJa bDWFJH"
>
  <header
    className="sc-bwzfXH huTMRj"
  >
    Header Part
  </header>
  <section
    className="sc-htpNat bsbVPk"
  >
    App
  </section>
  <footer
    className="sc-bxivhb gRxwYG"
  >
    Footer
  </footer>
</div>
`;
